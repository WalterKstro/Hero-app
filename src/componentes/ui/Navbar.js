import { useContext } from "react";
import { NavLink, useHistory } from "react-router-dom";
import AuthContext from "../../auth/AuthContext";

const Navbar = () => {
    const {user, dispatch} = useContext(AuthContext)
    const history = useHistory()
    
    const handleLogout = () =>{
        dispatch({
            type: 'logout'
        })
        history.replace('/login')
    }
    return (
        <nav className="sticky-top navbar navbar-expand-lg navbar-dark bg-dark py-0">
            <div className="container-fluid">
                <NavLink to="/" className="navbar-brand"><img src={`${process.env.PUBLIC_URL}/assets/Brand.svg`} alt="Brand"/></NavLink>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink activeClassName="active" to="/dc" className="nav-link">DC Universe</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink activeClassName="active" to="/marvel" className="nav-link">Marvel Universe</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink activeClassName="active" to="/search" className="nav-link">Search</NavLink>
                        </li>
                    </ul>
                </div>

                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav ms-auto">
                        <li className="nav-item">
                            <button className="nav-link text-white fw-bold btn">
                                {user.email}
                            </button>
                        </li>
                        <li className="nav-item">
                            <button 
                                className="btn btn-primary"
                                onClick={handleLogout}
                                >
                                    Sign Out
                            </button>
                        </li>
                    </ul>
                </div>
            </div>
        </nav>
    )
}

export default Navbar
