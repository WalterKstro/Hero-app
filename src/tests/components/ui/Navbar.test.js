import { mount } from "enzyme"
import { MemoryRouter, Router } from "react-router"
import AuthContext from "../../../auth/AuthContext"
import Navbar from "../../../componentes/ui/Navbar"

describe('Testing in <Navbar/>', () => {

    const user = {
        email: 'Walter Francisco',
        pass: '123'
    }
    const dispatch = jest.fn()

    const history = {
        replace: jest.fn(),
        length: 0,
        location: { 
            pathname: '',
            search: '',
            state: '',
            hash: ''
        },
        push: jest.fn(),
        go: jest.fn(),
        goBack: jest.fn(),
        goForward: jest.fn(),
        block: jest.fn(),
        listen: jest.fn(),
        createHref: jest.fn()
    }
    
    let rendered
    beforeEach(() => {
        rendered = mount(
            <AuthContext.Provider value={{user, dispatch}}>
                <MemoryRouter>
                    <Router history={history}>
                        <Navbar/>
                    </Router>
                </MemoryRouter>
            </AuthContext.Provider>
        )
    })

    test('should show correctly', () => {
        expect(rendered.find('.text-white').text().trim()).toBe(user.email)
        expect(rendered).toMatchSnapshot()
    })

    test('should call function handleLogout,throw dispatch and use history for redirect to page login', () => {
        const button = rendered.find('.btn-primary')
        button.simulate('click')

        expect(dispatch).toBeCalled()
        expect(dispatch).toBeCalledTimes(1)
        expect(dispatch).toBeCalledWith({type: 'logout'})
        expect(history.replace).toBeCalledWith('/login')
    })
    
    
})
