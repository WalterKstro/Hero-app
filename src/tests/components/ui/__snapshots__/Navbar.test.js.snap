// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <Navbar/> should show correctly 1`] = `
<nav
  className="sticky-top navbar navbar-expand-lg navbar-dark bg-dark py-0"
>
  <div
    className="container-fluid"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Brand"
        src="/assets/Brand.svg"
      />
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            DC Universe
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Marvel Universe
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
      </ul>
    </div>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="nav-link text-white fw-bold btn"
          >
            Walter Francisco
          </button>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Sign Out
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
